conference/src/db/mod.rs

```diff
@@ -17,6 +19,38 @@ pub async fn create_pool(
         .expect("Failed to create sqlx database pool")
 }
 
+#[allow(non_camel_case_types)]
+pub enum AdvisoryLockId {
+    ASSIGN_ROOM_TO_BACKEND,
+}
+
+pub async fn acquire_assign_room_to_backend_lock<'c>(
+    trans: &mut sqlx::Transaction<'c, sqlx::Postgres>,
+) -> sqlx::Result<()> {
+    // NOTE: Advisory lock
+    // `SELECT pg_advisory_xact_lock(1);`
+    // was compared with table lock
+    // `LOCK TABLE room IN SHARE UPDATE EXCLUSIVE MODE;`
+    // and setting transaction isolation level
+    // `SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;`
+    // Starting with the latest, with isolation level serializable, we'd have to explicitly repeat the transaction.
+    // Also, it means it's the slowest way performance-wise.
+    // `LOCK TABLE room` doesn't have that downside but locks entire table that is also written to by different queries
+    // (not in this transaction).
+    // Alternative `LOCK TABLE janus_backend` (or locking any other table used in this transaction) would also work
+    // but it doesn't indicate the intent that we just want to serialize these transactions.
+    // I.e. locking _any_ table would work here - meaning we just need an unique lock.
+    // We don't rely on semantics of the lock apart from that we need an _exclusive_ (not shared) lock.
+    //
+    // Guiding tests are `connect_incorrect_balance_*`
+    let query = sqlx::query!(
+        "SELECT pg_advisory_xact_lock($1);",
```

Comment:
```suggestion
        "SELECT pg_advisory_xact_lock($1) /* assign_room_to_backend_lock */;",
```

Было бы удобно добавить комментарий к запросу, чтобы его можно было идентифицировать в списке текущих запросов в PG среди други локов, когда или если они появятся.

Как вараинт еще можно использовать не комментарий, а `SELECT pg_advisory_xact_lock($1) AS ...`.